src/NLog/Targets/FileTarget.cs

```diff
@@ -1759,50 +1623,85 @@ private byte[] GetLayoutBytes(Layout layout)
             return this.TransformBytes(this.Encoding.GetBytes(renderedText));
         }
 
-        private void InvalidateCacheItem(string fileName)
+#if !SILVERLIGHT
+        private static string CleanupInvalidFileNameChars(string fileName)
+        {
+            var lastDirSeparator =
+                fileName.LastIndexOfAny(new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
+
+            var fileName1 = fileName.Substring(lastDirSeparator + 1);
+            var dirName = lastDirSeparator > 0 ? fileName.Substring(0, lastDirSeparator) : string.Empty;
+            fileName1 = Path.GetInvalidFileNameChars().Aggregate(fileName1, (current, c) => current.Replace(c, '_'));
+            return Path.Combine(dirName, fileName1);
+        }
+#endif
+
+        private sealed class InitializedFiles
         {
-            for (int i = 0; i < this.recentAppenders.Length; ++i)
+            // Clean up Period is defined in days.
+            public const int CleanupPeriod = 2;
+
+            public const int MaxAllowed = 100;
```

Comment:
not sure if public (not really public of course) are a good sign. They should be private and otherwise there are at the wrong class I think.